go/oasis-node/cmd/node/node.go

```diff
@@ -526,7 +525,8 @@ func NewTestNode() (*Node, error) {
 	return newNode(true)
 }
 
-func newNode(testNode bool) (n *Node, err error) { // nolint: gocyclo
+func newNode(testNode bool) (*Node, error) { // nolint: gocyclo
+	var err error
```

Comment:
The reason for having that named `err` return value is for the deferred func below to propagate the error. This prevents that from happening. Maybe add a comment so this doesn't get changed again :-)

Reply:
Oh right, however in that case any objections to doing `(node *Node, err error)` ? (and not using a separate `n` variable for node here). Because otherwise having `n` in scope through the function which is always `nil` is a bit dangerous :grin: that's also what lead me to removing the named returns here. The only "downside" will be that a non-nil `node`  will be returned in case this function errors, but i think that's fine as?

Reply:
So we did have it like that but changed it in d0551abb74b878bf62b44115fd917a9786c9357b due to a "possible nil dereference", but can't currently think of a case where this would be problematic.

Reply:
hm, so either whatever triggered that was fixed already (or it was a [false positive](https://github.com/dominikh/go-tools/issues/641)) since it doesn't seem to happen anymore

Reply:
ah i did found it, so `node` was `nil` in the deferred function. Fixed it by passing `node` into the deferred function via an argument